examples/notebooks/beam-ml/run_inference_xgboost.ipynb

```diff
@@ -0,0 +1,374 @@
+        "    iris_dataset['data'], iris_dataset['target'], test_size=.2, random_state=SEED)\n",
+        "booster = xgboost.XGBClassifier(\n",
+        "    n_estimators=2, max_depth=2, learning_rate=1, objective='binary:logistic')\n",
+        "booster.fit(x_train_classification, y_train_classification)\n",
+        "booster.save_model(CLASSIFICATION_MODEL_STATE)\n",
+        "\n",
+        "# Train the regression model\n",
+        "california_dataset = fetch_california_housing()\n",
+        "x_train_regression, x_test_regression, y_train_regression, y_test_regression = train_test_split(\n",
+        "    california_dataset['data'], california_dataset['target'], test_size=.2, random_state=SEED)\n",
+        "model = xgboost.XGBRegressor(\n",
+        "    n_estimators=1000,\n",
+        "    max_depth=8,\n",
+        "    eta=0.1,\n",
+        "    subsample=0.75,\n",
+        "    colsample_bytree=0.8)\n",
+        "model.fit(x_train_regression, y_train_regression)\n",
+        "model.save_model(REGRESSION_MODEL_STATE)\n",
+        "\n",
+        "\n",
+        "# Reshape the test data as XGBoost expects a batch instead of a single element\n",
+        "# More information: https://xgboost.readthedocs.io/en/stable/prediction.html\n",
+        "x_test_classification = x_test.reshape(5, 6, 4)\n",
+        "x_test_regression = x_test_regression.reshape(258, 16, 8)"
+      ],
+      "metadata": {
+        "id": "KVSKt3pFPBnj"
+      },
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "source": [
+        "### Postprocessing helper functions"
+      ],
+      "metadata": {
+        "id": "VGQj-B1Abioq"
+      }
+    },
+    {
+      "cell_type": "code",
+      "source": [
+        "def translate_labels(inference_results: PredictionResult):\n",
+        "  \"\"\"\n",
+        "    Maps output values (0, 1 or 2) of the XGBoost Iris classification\n",
+        "    model to the names of the different Iris flowers.\n",
+        "    Args:\n",
+        "      inference_results: Array containing the outputs of the XGBoost Iris classification model\n",
+        "    \"\"\"\n",
+        "  return PredictionResult(\n",
+        "      inference_results.example,\n",
+        "      np.vectorize(['Setosa', 'Versicolour',\n",
+        "                    'Virginica'].__getitem__)(inference_results.inference))\n",
+        "\n",
+        "\n",
+        "class FlattenBatchPredictionResults(beam.DoFn):\n",
+        "  \"\"\"This function takes a batch (list) of\n",
+        "  PredictionResults as input and yield all elements\"\"\"\n",
+        "  def process(self, batch_prediction_result: PredictionResult):\n",
+        "    for example, inference in zip(batch_prediction_result.example, batch_prediction_result.inference):\n",
+        "      yield PredictionResult(\n",
+        "          example, inference, batch_prediction_result.model_id)\n"
```

Comment:
Instead of yielding the result, how about we just print it here (rather than having an extra map to do that everywhere)?